### PR TITLE
fix: Update second Allow location access modal on web

### PR DESCRIPTION
**Pull request by truph01**
### Fixed Issues

$ https://github.com/Expensify/App/issues/50601
PROPOSAL: https://github.com/Expensify/App/issues/50601#issuecomment-2406343329

### Tests

Prerequisite for all test cases: 

Update code:

https://github.com/Expensify/App/blob/61ea52bce121d57b6ce379f37895735b3602c385/src/pages/iou/request/step/IOURequestStepConfirmation.tsx#L591-L601

````javascript
        if (gpsRequired) {
            setStartLocationPermissionFlow(true);
            return;
        }
````

to remove the 7 days period.

Test 1: For [Chrome]

1. Clear all the ND's permission in chrome://settings/content/location if has.
2. FAB > Submit expense > Scan > Upload receipt.
3. In confirmation page, click "Submit expense". Verify the below modal is displayed:

image

4. Choose "Continue". Verify the below modal is displayed:

image

5. Choose "Block".

6. FAB > Submit expense > Scan > Upload receipt.

7. In confirmation page, click "Submit expense". Verify the below modal is displayed:

image

8. Choose "Got it". Verify the expense is submitted.

Test 2: For [Safari]

1. Login in ND in new tab.
2. FAB > Submit expense > Scan > Upload receipt.
3. In confirmation page, click "Submit expense". Verify the below modal is displayed:

image

4. Choose "Continue". Verify the below modal is displayed:

image

5. Choose "Don't allow".

6. FAB > Submit expense > Scan > Upload receipt.

7. In confirmation page, click "Submit expense". Verify the below modal is displayed:

image

8. Choose "Continue". Verify the expense is submitted.

Test 3: For Android/IOS app

1. FAB > Submit expense > Scan > Upload receipt.
2. In confirmation page, click "Submit expense". Verify the below modal is displayed:

image

3. Choose "Continue". Verify the below modal is displayed:

image

4. Choose "Don't Allow".

5. FAB > Submit expense > Scan > Upload receipt.

6. In confirmation page, click "Submit expense". Verify the below modal is displayed:

image

7. Choose "Settings". Verify app redirect to setting screen.

- [x] Verify that no errors appear in the JS console

### QA Steps

Test 1: For [Chrome]

1. Clear all the ND's permission in chrome://settings/content/location if has.
2. FAB > Submit expense > Scan > Upload receipt.
3. In confirmation page, click "Submit expense". Verify the below modal is displayed:

image

4. Choose "Continue". Verify the below modal is displayed:

image

5. Choose "Block".

6. 7 days later: FAB > Submit expense > Scan > Upload receipt.

7. In confirmation page, click "Submit expense". Verify the below modal is displayed:

image

8. Choose "Got it". Verify the expense is submitted.

Test 2: For [Safari]

1. Login in ND in new tab.
2. FAB > Submit expense > Scan > Upload receipt.
3. In confirmation page, click "Submit expense". Verify the below modal is displayed:

image

4. Choose "Continue". Verify the below modal is displayed:

image

5. Choose "Don't allow".

6. 7 days later: FAB > Submit expense > Scan > Upload receipt.

7. In confirmation page, click "Submit expense". Verify the below modal is displayed:

image

8. Choose "Continue". Verify the expense is submitted.

Test 3: For Android/IOS app (This PR does not change the Android/IOS logic, so just add the test steps to make sure it works as it is)

1. FAB > Submit expense > Scan > Upload receipt.
2. In confirmation page, click "Submit expense". Verify the below modal is displayed:

image

3. Choose "Continue". Verify the below modal is displayed:

image

4. Choose "Don't Allow".

5. 7 days later: FAB > Submit expense > Scan > Upload receipt.

6. In confirmation page, click "Submit expense". Verify the below modal is displayed:

image

7. Choose "Settings". Verify app redirect to setting screen.
- [x] Verify that no errors appear in the JS console

### PR Author Checklist

- [x] I linked the correct issue in the `### Fixed Issues` section above
- [x] I wrote clear testing steps that cover the changes made in this PR
    - [x] I added steps for local testing in the `Tests` section
    - [x] I added steps for the expected offline behavior in the `Offline steps` section
    - [x] I added steps for Staging and/or Production testing in the `QA steps` section
    - [x] I added steps to cover failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [x] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
    - [x] I tested this PR with a [High Traffic account](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#high-traffic-accounts) against the staging or production API to ensure there are no regressions (e.g. long loading states that impact usability).
- [x] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I ran the tests on **all platforms** & verified they passed on:
    - [x] Android: Native
    - [x] Android: mWeb Chrome
    - [x] iOS: Native
    - [x] iOS: mWeb Safari
    - [x] MacOS: Chrome / Safari
    - [x] MacOS: Desktop
- [x] I verified there are no console errors (if there's a console error not related to the PR, report it or open an issue for it to be fixed)
- [x] I followed proper code patterns (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [x] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`)
    - [x] I verified that the left part of a conditional rendering a React component is a boolean and NOT a string, e.g. `myBool && <MyComponent />`.
    - [x] I verified that comments were added to code that is not self explanatory
    - [x] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [x] I verified any copy / text shown in the product is localized by adding it to `src/languages/*` files and using the [translation method](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60)
      - [x] If any non-english text was added/modified, I verified the translation was requested/reviewed in #expensify-open-source and it was approved by an internal Expensify engineer. Link to Slack message:
    - [x] I verified all numbers, amounts, dates and phone numbers shown in the product are using the [localization methods](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60-L68)
    - [x] I verified any copy / text that was added to the app is grammatically correct in English. It adheres to proper capitalization guidelines (note: only the first word of header/labels should be capitalized), and is either coming verbatim from figma or has been approved by marketing (in order to get marketing approval, ask the Bug Zero team member to add the Waiting for copy label to the issue)
    - [x] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [x] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [x] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [x] I followed the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [x] I tested other components that can be impacted by my changes (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` are working as expected)
- [x] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [x] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [x] I verified that if a function's arguments changed that all usages have also been updated correctly
- [x] If any new file was added I verified that:
    - [x] The file has a description of what it does and/or why is needed at the top of the file if the code is not self explanatory
- [x] If a new CSS style is added I verified that:
    - [x] A similar style doesn't already exist
    - [x] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/styles/utils/index.ts) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(theme.componentBG)`)
- [x] If the PR modifies code that runs when editing or sending messages, I tested and verified there is no unexpected behavior for all supported markdown - URLs, single line code, code blocks, quotes, headings, bold, strikethrough, and italic.
- [x] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [x] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [x] If the PR modifies a component or page that can be accessed by a direct deeplink, I verified that the code functions as expected when the deeplink is used - from a logged in and logged out account.
- [x] If the PR modifies the UI (e.g. new buttons, new UI components, changing the padding/spacing/sizing, moving components, etc) or modifies the form input styles:
    - [x] I verified that all the inputs inside a form are aligned with each other.
    - [x] I added `Design` label and/or tagged `@Expensify/design` so the design team can review the changes.
- [x] If a new page is added, I verified it's using the `ScrollView` component to make it scrollable when more elements are added to the page.
- [x] If the `main` branch was merged into this PR after a review, I tested again and verified the outcome was still expected according to the `Test` steps.

### Screenshots/Videos
<details>
<summary>Android: Native</summary>

https://github.com/user-attachments/assets/10657a86-ef59-4399-be37-321856d3e926

</details>

<details>
<summary>Android: mWeb Chrome</summary>

https://github.com/user-attachments/assets/f0787700-0661-4e9a-bc68-2bde717db411

</details>

<details>
<summary>iOS: Native</summary>

https://github.com/user-attachments/assets/ca4c9440-7bf7-46b8-8c61-8fad9c4f362d

</details>

<details>
<summary>iOS: mWeb Safari</summary>

https://github.com/user-attachments/assets/c29dd444-d110-4b6c-b4d3-7d77e880ee08

</details>

<details>
<summary>MacOS: Chrome / Safari</summary>

https://github.com/user-attachments/assets/3a5a0de1-0f67-4acc-9c5f-f97106821c06

</details>

<details>
<summary>MacOS: Desktop</summary>

https://github.com/user-attachments/assets/403a0b7f-d953-4744-afaf-06c8b1bf4c58

</details>

### CODE REVIEW

**Comment by alitoshmatov** [comment]
@truph01 After not allowing location access the modal is not appearing in second submit. Maybe it is because I am in newer version of chrome where it offer different options for choosing allowing access


https://github.com/user-attachments/assets/6637ce83-ccd8-480b-9cad-add27a512d4c

**Comment by truph01** [comment]
@alitoshmatov Ah, I see it. Do you think it's actually related to our PR, given that we're focusing on updating the modal (title, subtitle, buttons) rather than modifying the logic for when it should be displayed?

**Comment by alitoshmatov** [comment]
> @alitoshmatov Ah, I see it. Do you think it's actually related to our PR, given that we're focusing on updating the modal (title, subtitle, buttons) rather than modifying the logic for when it should be displayed?

I guess not, but would be great if we considered this possibility as well.

**Comment by alitoshmatov** [comment]
@truph01 I am also noticing that in your ios safari recording, the modal is shown second time you submit a scan but clicking continue does not show native location permission modal. In my simulator after I reject location first time, no modal is shown in second scan request

https://github.com/user-attachments/assets/adbd09d7-e593-415a-afa4-c30d8e660cb2

**Comment by alitoshmatov** [comment]
In ios also the modal is shown only in first scan request. After rejecting native permission modal first time submitting second invoice request is not triggering modal


https://github.com/user-attachments/assets/44edd09a-b14b-4175-92ca-dd1a6b5afaa6

**Comment by alitoshmatov** [comment]
That is three places – Ios native, Ios safari, Mac chrome which shows only first modal not the second modal with "Got it" button or "Settings" button.

If this is accepted behavior we should reconsider testing steps to prevent further confusion

**Comment by truph01** [comment]
We are encountering different behavior in Chrome, maybe because of the Chrome version ([Here](https://github.com/Expensify/App/pull/51709#issuecomment-2453493756) is from @alitoshmatov and [the one in the screen recording section](https://github.com/Expensify/App/pull/51709#issue-2622631288) is from my side). So, I wonder how I can write the test steps properly in Chrome?


 cc @Julesssss

**Comment by Julesssss** [comment]
> @alitoshmatov Ah, I see it. Do you think it's actually related to our PR, given that we're focusing on updating the modal (title, subtitle, buttons) rather than modifying the logic for when it should be displayed?

> In ios also the modal is shown only in first scan request. After rejecting native permission modal first time submitting second invoice request is not triggering modal

I think that sounds like [our logic](https://github.com/Expensify/App/pull/48237) that only prompts the user one time a week? 

> 4. Choose "Don't Allow".

This screenshot is incorrect for mobile, can you change it to the native permission modal so that QA aren't confused.

**Comment by truph01** [comment]
> retesting with the 7 day throttle disabled

@alitoshmatov The behaviors in [here](https://github.com/Expensify/App/pull/51709#issuecomment-2453493756), [here](https://github.com/Expensify/App/pull/51709#issuecomment-2457845387) and [here](https://github.com/Expensify/App/pull/51709#issuecomment-2457890342) are tested when disabling the 7 day throttle, no? If no, could you help try again?

**Comment by alitoshmatov** [comment]
Same result is happening, I think branched test PR does not have correct changes
```diff
-            const shouldStartLocationPermissionFlow =
-                !lastLocationPermissionPrompt ||
-                (DateUtils.isValidDateString(lastLocationPermissionPrompt ?? '') &&
-                    DateUtils.getDifferenceInDaysFromNow(new Date(lastLocationPermissionPrompt ?? '')) > CONST.IOU.LOCATION_PERMISSION_PROMPT_THRESHOLD_DAYS);
+            const shouldStartLocationPermissionFlow = !lastLocationPermissionPrompt;
```

The condition is not what we expect here. `shouldStartLocationPermissionFlow` is `false` in the second try because after first time `lastLocationPermissionPrompt` stores the timestamp of that.

**Comment by alitoshmatov** [comment]
I think `shouldStartLocationPermissionFlow` should be always true to ignore 7 day throttle. 

Also, @truph01  if there were 7 day throttle how were you able to test your PR and produce expected results(two permission modals sequentially) since it is clear that chrome version was not playing any role here.

**Comment by truph01** [comment]
> Also, @truph01 if there were 7 day throttle how were you able to test your PR and produce expected results

I forgot to mention that I also temporarily removed the 7-day throttle as a hard-coded adjustment for testing purposes.

**Comment by truph01** [comment]
@alitoshmatov Could you give me the Chrome version from your side? 

I updated the version but cannot reproduce like [this one](https://github.com/Expensify/App/pull/51709#issuecomment-2453493756):

image

**Comment by alitoshmatov** [comment]
Now after setting `shouldStartLocationPermissionFlow` to always `true` everything works as expected. Well kinda, I don't know if followings are expected



1. In android in second attempt modal is showing `Continue` button after it is pressed then it is switching to `Settings`(see video 👇 )
2. In ios safari, second attemp modal is showing `Continue` button and pressing it does nothing and submits the scan


https://github.com/user-attachments/assets/a029638e-d7f5-4190-a751-79106499fd42

https://github.com/user-attachments/assets/195bdefa-1a41-41a3-8915-3a7f14672245

**Comment by alitoshmatov** [comment]
> @alitoshmatov Could you give me the Chrome version from your side?

@truph01 As I said [here](https://github.com/Expensify/App/pull/51709#issuecomment-2466680771), branched PR by @Julesssss had wrong conditions to ignore 7 day period that is why I got the same result. Now after correctly removing 7 day throttle issue is solved in web chrome.

**Comment by Julesssss** [comment]
_Just a note, iOS only lets us prompt once for location permission if users says NO. But Android lets us prompt twice. So there is a slight difference._

**Android**
> In android in second attempt modal is showing Continue button after it is pressed then it is switching to Settings(see video 👇 )

This is almost correct I think:

**1st attempt - 'Dont' allow'**
- Expense submitted

**2nd attempt - 'Dont' allow'**
- Expense wasn't submitted -- ❌ it should have submitted without location, that is the only bug I think
- Button changes to settings -- ✅ user shouldn't see this, but it is technically correct, after the second 'Not now' press the button will prompt users to manually enable in settings 

**iOS**
> In ios safari, second attempt modal is showing Continue button and pressing it does nothing and submits the scan

I think this is expected. But that's not what I see in your video -- I see the native app:

Screenshot 2024-11-12 at 10 53 04

**Comment by alitoshmatov** [comment]
@Julesssss Sorry uploaded wrong file, here is ios safari version

https://github.com/user-attachments/assets/5e290c15-8ba1-4520-a95a-2421f1a68a65

**Comment by Julesssss** [comment]
Thanks @alitoshmatov

> In ios safari, second attemp modal is showing Continue button and pressing it does nothing and submits the scan

Cool so this seems like the last thing to fix. We should be showing the updated text here for web

@truph01 the test steps (and desired behaviour) is that we show this different text for web on the second attempt, do you think maybe a recent change broken that? Once that's showing again I think we are done here

 
381365242-8d43f39b-51b7-4bbe-857e-0c5e3aea0590

File: src/components/LocationPermissionModal/index.tsx [hunk 1]
```diff
@@ -81,15 +81,22 @@ function LocationPermissionModal({startPermissionFlow, resetPermissionFlow, onDe
         setShowModal(false);
         resetPermissionFlow();
     };
+
+    const locationErrorMessage = useMemo(() => (isWeb ? 'receipt.allowLocationFromSetting' : 'receipt.locationErrorMessage'), [isWeb]);
```
Comment (Julesssss) [inline src/components/LocationPermissionModal/index.tsx:85]: We don't seem to show this message on web with the current changes

**Comment by truph01** [comment]
@alitoshmatov The behavior in safari (both MacOS and IOS) is:

1. Login in the incognito tab.
2. FAB > Submit expense > Scan > Upload receipt.
3. In confirmation page, click "Submit expense". Verify the below modal is displayed (It is a pre-check current permission step and the permission returned is "Denied"):

image


4. Choose "Continue". Verify the below modal is displayed (It is a prompt step):

image


5. Choose "Don't allow". (The permission returned is "Blocked")

6. FAB > Submit expense > Scan > Upload receipt.

7. On confirmation page, click "Submit expense". Verify the below modal is displayed (It is a pre-check current permission step and the permission returned is "Denied"):

image


8. Choose "Continue". (Now the permission returned is "Blocked"). Expense is submitted.



In Chrome, we check whether the pre-check for the current permission status returns "Blocked." This allows us to determine if the previous prompt was rejected. If it was, we display the second "Allow location access" modal.

In Safari, however, the pre-check for the current permission status always returns "Denied," which makes it impossible to determine if the previous prompt was rejected. Consequently, it becomes difficult to decide when to display the second "Allow location access" modal. As a result, Safari always shows the first modal. This discrepancy leads to the observed behavior.

> In ios safari, second attemp modal is showing Continue button and pressing it does nothing and submits the scan

**Comment by alitoshmatov** [comment]
> In Safari, however, the pre-check for the current permission status always returns "Denied,"

I see. I think everything looks good, @truph01 please update QA steps for each case in OP. I will complete the checklist

**Comment by truph01** [comment]
@alitoshmatov I updated test cases for Chrome and Safari.

**Comment by alitoshmatov** [comment]
## Reviewer Checklist

- [x] I have verified the author checklist is complete (all boxes are checked off).
- [x] I verified the correct issue is linked in the `### Fixed Issues` section above
- [x] I verified testing steps are clear and they cover the changes made in this PR
    - [x] I verified the steps for local testing are in the `Tests` section
    - [x] I verified the steps for Staging and/or Production testing are in the `QA steps` section
    - [x] I verified the steps cover any possible failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [x] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
- [x] I checked that screenshots or videos are included for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I verified tests pass on **all platforms** & I tested again on:
    - [x] Android: Native
    - [x] Android: mWeb Chrome
    - [x] iOS: Native
    - [x] iOS: mWeb Safari
    - [x] MacOS: Chrome / Safari
    - [x] MacOS: Desktop
- [x] If there are any errors in the console that are unrelated to this PR, I either fixed them (preferred) or linked to where I reported them in Slack
- [x] I verified proper code patterns were followed (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [x] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`).
    - [x] I verified that the left part of a conditional rendering a React component is a boolean and NOT a string, e.g. `myBool && <MyComponent />`.
    - [x] I verified that comments were added to code that is not self explanatory
    - [x] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [x] I verified any copy / text shown in the product is localized by adding it to `src/languages/*` files and using the [translation method](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60)
    - [x] I verified all numbers, amounts, dates and phone numbers shown in the product are using the [localization methods](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60-L68)
    - [x] I verified any copy / text that was added to the app is grammatically correct in English. It adheres to proper capitalization guidelines (note: only the first word of header/labels should be capitalized), and is either coming verbatim from figma or has been approved by marketing (in order to get marketing approval, ask the Bug Zero team member to add the Waiting for copy label to the issue)
    - [x] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [x] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [x] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [x] I verified that this PR follows the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [x] I verified other components that can be impacted by these changes have been tested, and I retested again (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` have been tested & I retested again)
- [x] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [x] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [x] If a new component is created I verified that:
    - [x] A similar component doesn't exist in the codebase
    - [x] All props are defined accurately and each prop has a `/** comment above it */`
    - [x] The file is named correctly
    - [x] The component has a clear name that is non-ambiguous and the purpose of the component can be inferred from the name alone
    - [x] The only data being stored in the state is data necessary for rendering and nothing else
    - [x] For Class Components, any internal methods passed to components event handlers are bound to `this` properly so there are no scoping issues (i.e. for `onClick={this.submit}` the method `this.submit` should be bound to `this` in the constructor)
    - [x] Any internal methods bound to `this` are necessary to be bound (i.e. avoid `this.submit = this.submit.bind(this);` if `this.submit` is never passed to a component event handler like `onClick`)
    - [x] All JSX used for rendering exists in the render method
    - [x] The component has the minimum amount of code necessary for its purpose, and it is broken down into smaller components in order to separate concerns and functions
- [x] If any new file was added I verified that:
    - [x] The file has a description of what it does and/or why is needed at the top of the file if the code is not self explanatory
- [x] If a new CSS style is added I verified that:
    - [x] A similar style doesn't already exist
    - [x] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/utils/index.ts) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(theme.componentBG`)
- [x] If the PR modifies code that runs when editing or sending messages, I tested and verified there is no unexpected behavior for all supported markdown - URLs, single line code, code blocks, quotes, headings, bold, strikethrough, and italic.
- [x] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [x] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [x] If the PR modifies a component or page that can be accessed by a direct deeplink, I verified that the code functions as expected when the deeplink is used - from a logged in and logged out account.
- [x] If the PR modifies the UI (e.g. new buttons, new UI components, changing the padding/spacing/sizing, moving components, etc) or modifies the form input styles:
    - [x] I verified that all the inputs inside a form are aligned with each other.
    - [x] I added `Design` label and/or tagged `@Expensify/design` so the design team can review the changes.
- [x] If a new page is added, I verified it's using the `ScrollView` component to make it scrollable when more elements are added to the page.
- [x] If the `main` branch was merged into this PR after a review, I tested again and verified the outcome was still expected according to the `Test` steps.
- [x] I have checked off every checkbox in the PR reviewer checklist, including those that don't apply to this PR.

### Screenshots/Videos
<details>
<summary>Android: Native</summary>


https://github.com/user-attachments/assets/8376d78e-7e91-4445-8c4e-a962e1570a57

</details>

<details>
<summary>Android: mWeb Chrome</summary>



https://github.com/user-attachments/assets/9ee1daf8-c02a-4b7c-ac9a-69cbcbfef46c

</details>

<details>
<summary>iOS: Native</summary>

https://github.com/user-attachments/assets/b7563d8e-c679-42ca-a867-69d6195f9619



</details>

<details>
<summary>iOS: mWeb Safari</summary>

https://github.com/user-attachments/assets/9a53468e-7d14-44c9-a5aa-6e3d001a9f15


</details>

<details>
<summary>MacOS: Chrome / Safari</summary>



https://github.com/user-attachments/assets/f7c6fafb-1d68-42a7-9090-ffc131b4b912

</details>

<details>
<summary>MacOS: Desktop</summary>


https://github.com/user-attachments/assets/90c7aa0c-0383-4e05-9e04-42bdb09a0e86


</details>

**Review by alitoshmatov — APPROVED** [review]
LGTM!

**Review by Julesssss — APPROVED** [review]
Thanks for the detailed tests. It's working well for me. 

I wasn't able to enable camera permission on Android mWeb but this already exists in production.

**Comment by Julesssss** [comment]
This PR didn't reduce perf by 19% -- we made simple modifications to the location permission flow for web.

**Comment by hannojg** [comment]
yes, thats a false positive, sorry

**Pull request merged**